catalogs/usfimr-s1/ingest_s1.py

```diff
@@ -0,0 +1,165 @@
+        )
+
+        search_results = search_sentinelhub_s1(
+            temporal_bounds_string, flood_bounds, http_pool, token
+        )
+        search_results = json.loads(search_results.data.decode("utf-8"))
+
+        result_count = search_results["context"]["returned"]
+        if result_count > 0:
+            flood_with_results.append(flood)
+            print(
+                "{result_count} results for USFIMR ID={flood_id}".format(
+                    result_count=result_count, flood_id=flood.id
+                )
+            )
+            # Estimate sq km of coverage
+            print(
+                "km2 of labelled region (estimated from convex hull of labels)",
+                area(flood.geometry) / 1000,
+            )
+            print("km2 of flooding", flood.properties["Flood_km2"])
+
+        else:
+            print(
+                "No S1 results found for USFIMR ID={flood_id}".format(flood_id=flood.id)
+            )
+
+    s3 = boto3.resource("s3")
+    bucket = s3.Bucket("noaafloodmapping-sentinelhub-batch-eu-central-1")
+
+    for i, flood in enumerate(flood_with_results):
+
+        # Check if data is already loaded
+        filtered_bucket_contents = bucket.objects.filter(
+            Prefix="glofimr/{}".format(flood.id)
+        )
+        already_loaded = False
+        for key in filtered_bucket_contents:
+            already_loaded = True
+            break
+
+        if already_loaded:
+            print("Flood {} already ingested, skipping".format(flood.id))
+        else:
+            print("Flood {} not ingested, ingesting".format(flood.id))
+            # temporal bounds
+            date_min, date_max = get_flood_temporal_bounds(flood)
+            temporal_bounds_string = (
+                date_min.isoformat() + "Z/" + date_max.isoformat() + "Z"
+            )
+            batch_creation_request = create_batch_request(
+                flood, temporal_bounds_string, http_pool, token
+            )
+            # from IPython import embed; embed()
+            batch_creation_response = json.loads(
+                batch_creation_request.data.decode("utf-8")
+            )
+            batch_request_id = batch_creation_response["id"]
+
+            analyze_batch_request(batch_request_id, http_pool, token)
+
+            initiate_batch_request(batch_request_id, http_pool, token)
+
+            for count in range(100):
+                status_response = check_batch_status(batch_request_id, http_pool, token)
+                status_response_data = json.loads(status_response.data.decode("utf-8"))
+                status = status_response_data["status"]
+
+                if status == "DONE":
+                    print(
+                        "Flood ingest for ID {} completed successfully.".format(
+                            flood.id
+                        )
+                    )
+                    print("SUCCESSFUL REQUEST ID: {}".format(batch_request_id))
```

Comment:
Might be helpful to print the full s3 path that results were written to for easier parsing